dotnet-build.sh

```diff
@@ -65,12 +91,19 @@ echo "building $CS_PROJECT_FILE"
 dotnet build -c $RELEASE --no-restore $CS_PROJECT_FILE
 
 # run tests
-echo "running tests"
-for f in *.Test/*.csproj
-do
-  echo "Processing $f file..."
-  dotnet test "$f" --no-restore -c $RELEASE -r "$DIST/test-results"  
-done
+if [ $WITH_TESTING ]
+then
+  echo "running tests"
+  for f in *.Test/*.csproj
+  do
+    echo "Processing $f file..."
+    dotnet test "$f" -c $RELEASE -r "$DIST/test-results" --filter "TestCategory!=[Integration]" /p:CollectCoverage=true /p:CoverletOutputFormat="opencover" /p:CoverletOutput="$DIST/test-coverlet"
+  done
+    
+  # generate coverage report
+  dotnet tool install dotnet-reportgenerator-globaltool --tool-path "$DIST/tools"
```

Comment:
We should include this in the container build. (furthermore, because it's version dependent)

Reply:
Good idea